### PR TITLE
Deployment preparation/balancer relayer v6

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Deployment preparation: Batch Relayer V6

Build-info: https://github.com/balancer/balancer-v2-monorepo/commit/d4c6742704337dcc721c6248e91f37e3af79e0cd (master branch)

Adds recovery mode exit and a new query version of multicall.

## Deployment preparation tasks checklist:

- [X] Deployments `README` is updated with the new task name and link
- [X] Task `README` has a description and links to the artifacts
- [X] `build-info` is updated with artifacts compiled at the specified commit
- [X] Artifacts are generated from `build-info`
- [X] New task has a working fork test 

## Deprecating old tasks checklist: 

- [X] Old task was moved to `deprecated/` directory
- [X] Deployments `README` is updated: old task is moved to the 'deprecated' section, and links are updated with the `deprecated/` prefix
- [X] Old task imports in `index.ts`, `input.ts` and fork tests are updated
- [X] Old task `README` is updated with a warning sign, a link to the replacement task and a short description 

## Code checklist:

- [x] The diff is legible and has no extraneous changes (upgrades hardhat-vyper to 3.0.5; could be separated if it's worth it)
- [X] Complex code has been commented in tests and scripts
- [X] Fork test block number and test cases are appropriate for this deployment
- [X] The base branch is either `master`, or there's a description of how to merge

### CODE REVIEW

File: tasks/20231031-batch-relayer-v6/test/test.fork.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import { BigNumberish } from '@helpers/numbers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { WeightedPoolEncoder } from '@helpers/models/pools/weighted/encoder';
+import { MAX_UINT256 } from '@helpers/constants';
+import { defaultAbiCoder } from '@ethersproject/abi/lib/abi-coder';
+import { describeForkTest, impersonate, getForkedNetwork, Task, TaskMode } from '@src';
+import * as expectEvent from '@helpers/expectEvent';
+import { sharedBeforeEach } from '@helpers/sharedBeforeEach';
+
+describeForkTest('BatchRelayerLibrary', 'mainnet', 15485000, function () {
```
Comment (jubeira) [inline tasks/20231031-batch-relayer-v6/test/test.fork.ts:13]: Can we use a newer block here?

File: tasks/20231031-batch-relayer-v6/test/test.fork.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          kind: TRANSFER_EXTERNAL_USER_BALANCE_OP_KIND,
+          asset: gauge,
+          amount: 1, // The amount must be non-zero to trigger the token transfer and liquidity limit update
+          sender: sender.address,
+          recipient: sender.address,
+        })),
+        0, // value: 0 (no need to transfer ETH)
+        [], // no output references
+      ]),
+    ]);
```
Comment (jubeira) [inline tasks/20231031-batch-relayer-v6/test/test.fork.ts:169]: This test could use some rework. We have a helper now that does this in `GaugeActions` (`gaugeCheckpoint`) which does this but it's much simpler to call. It should work both in mainnet and the L2s.

**Review by jubeira — COMMENTED** [review]
Thanks for putting this together @EndymionJkb !
I have a few comments about the configuration and the gauge checkpoint feature which is also included in this release.

I'll review the wrapper tests in a second pass, but it might just be OK to keep the same block numbers as in the previous release.

File: tasks/20231031-batch-relayer-v6/test/composableStableV1.fork.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  cacheDurations,
+  exemptFlags,
+  rateProviders,
+  swapFeePercentage,
+  tokens,
+  initialBalances,
+  PoolKind,
+} from './helpers/sharedStableParams';
+
+describeForkTest('BatchRelayerLibrary - Composable Stable V1', 'mainnet', 16083775, function () {
```
[the remark this answers is not in the document]
Comment (EndymionJkb) [reply tasks/20231031-batch-relayer-v6/test/composableStableV1.fork.ts:24]: Seems to depend on old state (wstETH, etc.)

File: tasks/20231031-batch-relayer-v6/test/test.fork.ts (hunk 2 above)
Comment (EndymionJkb) [reply tasks/20231031-batch-relayer-v6/test/test.fork.ts]: Did the mainnet one first.

**Review by jubeira — APPROVED** [review]
I've verified that the build-info matches with the one in https://github.com/balancer/balancer-v2-monorepo/commit/d4c6742704337dcc721c6248e91f37e3af79e0cd. LGTM, thanks!

**Pull request merged**